packages/plugin-ext/src/plugin/type-converters.ts

```diff
@@ -818,10 +818,10 @@ export function toTask(taskDto: TaskDto): theia.Task {
         result.execution = getShellExecution(taskDto);
     }
 
-    if (taskType === 'customExecution' || types.CustomExecution.is(execution)) {
-        result.execution = getCustomExecution(taskDto);
+    if (taskType === 'customExecution') {
```

Comment:
could you clarify why `types.CustomExecution.is(execution)` check was removed here
I see that such check is still used in other places, [here](https://github.com/eclipse-theia/theia/blob/dece3526c5257f1bcab110b68347d53a8eec8c71/packages/plugin-ext/src/plugin/type-converters.ts#L776), for example

Reply:
Task type can be exactly one of `'shell' | 'process' | 'customExecution'`, so looking at the shape of the executino object instead makes no sense. The example you link is of converting a theia.Task object to a task DTO, which not the same. You could argue that the test for the 'shell' type above should also be fixed, though.